lib/drivers/insuletDriver.js

```diff
@@ -1512,7 +1520,15 @@ module.exports = function (config) {
       postrecords = buildBGRecords(data, postrecords);
       // first sort by log index
       postrecords = _.sortBy(postrecords, function(d) {
-        return d.index;
+        if (d.index) {
+          return d.index;
+        }
+        else if (d.payload && d.payload.logIndices) {
+          return d.payload.logIndices[0];
+        }
+        else {
```

Comment:
I can't find anywhere that indicates that returning undefined from a sort comparator has well-defined behavior. What are you intending this to do?